### ISSUE TITLE
Set destination as navigation target

### ISSUE THREAD

**Issue by antonspace** [pioneerspacesim/pioneer#5517]
### Observed behaviour

Some missions, once completed, clicking the "Set destination as navigation target" button in the "Mission", then "more info" page will set the destination to where the mission reward is to be collected, in recent combat missions, it seems this button remains set to the location of the mission, while it is no big deal to search the map and find the final destination, the inconsistency is what is the issue here

Here is the savegame.
[skipjack9.gz](https://github.com/pioneerspacesim/pioneer/files/10557515/skipjack9.gz)

__My pioneer version (and OS):__

Linux.
20230131
![Screenshot_2023-02-01_13-25-10](https://user-images.githubusercontent.com/32020353/216058729-d93c826c-6d99-4390-a5e0-b1477c9069c5.png)

**Comment by impaktor** [comment]
You mean, it should switch back to the station where you're supposed to collect the payment?

**Comment by antonspace** [comment]
Yes, In this case payment was collected upon entering the system,

Another thing, it may be using flavour9, that is if I implemented it correctly.
